src/oom/oomd-manager.c

```diff
@@ -408,7 +408,7 @@ static int monitor_swap_contexts_handler(sd_event_source *s, uint64_t usec, void
                         log_debug_errno(r, "Failed to get monitored swap cgroup candidates, ignoring: %m");
 
                 threshold = m->system_context.swap_total * THRESHOLD_SWAP_USED_PERCENT / 100;
-                r = oomd_kill_by_swap_usage(candidates, threshold, m->dry_run, &selected);
+                r = oomd_kill_by_swap_usage(candidates, threshold, m->dry_run, &selected, &(struct PreKillContext){m->prekill_timeout, m->event, m->prekill_ctxs});
```

Comment:
I think `oomd_kill_by_swap_usage()` should be split up into two functions, one function `oomd_select_by_swap_usage()` which returns the `OomdCGroupContext` to kill, and `oomd_kill_by_swap_usage()` which only does the actual killing. Then you can handle the prekill logic outside of both these functions instead of having to squeeze it into oomd_kill_by_swap_usage() like you're doing here.